src/main/java/org/opensearch/knn/plugin/transport/GetModelRequest.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.plugin.transport;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+
+/**
+ * {@link GetModelRequest} gets Model for given modelID
+ */
+public class GetModelRequest extends ActionRequest {
+
+    private String modelID;
+    public GetModelRequest(String modelID) {
```

Comment:
Do we need to validate if modelID is null? I know we do this in Rest handler, but if someone makes the transport action call directly, this could lead to issues.

Reply:
Ack